module/User/src/Model/User.php

```diff
@@ -152,13 +152,29 @@ public function getRoleNames()
     }
 
     /**
-     * Get the user's role ID.
-     *
      * @return string
      */
-    public function getRoleId()
+    public function getRoleId(): string
     {
-        return 'user_' . $this->getLidnr();
+        if (in_array('admin', $this->getRoleNames())) {
```

Comment:
Perhaps it is worth it to not do `$this->getRoleNames()` for the role check (i.e. make it a variable and pass the variable to `in_array()`).